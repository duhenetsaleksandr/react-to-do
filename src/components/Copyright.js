import React from 'react';

export default function Copyright() {
    return (<div>Â©Copyright. All rights deserved.</div>);
}
