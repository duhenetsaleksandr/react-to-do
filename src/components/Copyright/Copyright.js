import React from 'react';
import { FormattedMessage } from 'react-intl';

const Copyright = () => (<><FormattedMessage id="copyright" /></>);

export default Copyright;
