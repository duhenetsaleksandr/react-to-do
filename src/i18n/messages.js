import { LOCALES } from './locales'

export const messages = {
    [LOCALES.ENGLISH]: {
        copyright: '©Copyright. All rights deserved.',
        button_add: 'Add task',
        button_edit: 'Edit task',
        placeholder: 'Enter new todo',
        not_todos: 'Not found todos!',
    },
    [LOCALES.RUSSIAN]: {
        copyright: '©Копирайт. Все права защищены.',
        button_add: 'Добавить',
        button_edit: 'Редактировать',
        placeholder: 'Введите новую задачу',
        not_todos: 'Задачи не найдены!',
    },
};
