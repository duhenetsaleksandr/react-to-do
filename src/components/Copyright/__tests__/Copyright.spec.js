import React from 'react';
import Copyright from '../Copyright';
import {IntlProvider} from "react-intl";
import {messages} from "i18n/messages";
import {LOCALES} from "i18n/locales";

describe('should render Copyright component', () => {
    let component;
    beforeEach(() => {
        component = mount(<Copyright/>, {
            wrappingComponent: IntlProvider,
            wrappingComponentProps: {
                messages: messages[LOCALES.ENGLISH],
                locale: LOCALES.ENGLISH,
                defaultLocale: LOCALES.ENGLISH,
            },
        });
    });

    it('should contain one div', () => {
        expect(component.find('div')).toHaveLength(1);
    });

    it('should contain text "copyright"', () => {
        const element = component.find('div');
        expect(element.getElement().props.children.props.id).toBe('copyright');
    });

    it('should render Copyright component snapshot', () => {
        const component = shallow(<Copyright/>);
        expect(component).toMatchSnapshot();
    });
});
