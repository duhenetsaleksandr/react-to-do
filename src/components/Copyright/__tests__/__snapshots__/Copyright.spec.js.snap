// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Copyright component should render Copyright component snapshot 1`] = `
<Fragment>
  <MemoizedFormattedMessage
    id="copyright"
  />
</Fragment>
`;
